src/commonMain/kotlin/com/soywiz/korma/geom/Matrix3D.kt

```diff
@@ -251,9 +251,14 @@ class Matrix3D {
         )
     }
 
-    inline fun setToRotation(angle: Angle, direction: Vector3D): Matrix3D = setToRotation(angle, direction.x, direction.y, direction.z)
-    inline fun setToRotation(angle: Angle, x: Number, y: Number, z: Number): Matrix3D = setToRotation(angle, x.toFloat(), y.toFloat(), z.toFloat())
-    fun setToRotation(angle: Angle, x: Float, y: Float, z: Float): Matrix3D {
+    inline fun setToRotation(angle: Angle, x: Float, y: Float, z: Float): Matrix3D = setToRotation(angle, Vector3D(x,y,z,0))
+    inline fun setToRotation(angle: Angle, x: Number, y: Number, z: Number): Matrix3D = setToRotation(angle, Vector3D(x,y,z,0))
+    fun setToRotation(angle: Angle, direction: Vector3D): Matrix3D {
```

Comment:
The non-inline overload should be the one using primitives:
`fun setToRotation(angle: Angle, x: Float, y: Float, z: Float): Matrix3D`

This is like that to avoid allocations, since you can call the primitive overload to optimize stuff. Or to provide a pre-allocated Vector3D.

Sadly on JVM project Vallhalla is still not out. JS doesn't have value types, and Kotlin Native doesn't support value types yet either.

So the other ones:
```
inline fun setToRotation(angle: Angle, x: Number, y: Number, z: Number): Matrix3D
fun setToRotation(angle: Angle, direction: Vector3D): Matrix3D 
```

The one with `Number` is there to allow mixing `Int`, `Float` and `Double` literals since in Kotlin literals are not automatically converted. The function is inline because `Number` would box on `JVM` and probably `Native` since it is a class. Doing it inline:

```
inline fun setToRotation(angle: Angle, x: Number, y: Number, z: Number): Matrix3D
setToRotation(90.degress, Integer.valueOf(1), Double.valueOf(0.0), Float.valueOf(3f)) // "Calling" the Number overload
setToRotation(90.degress, Integer.valueOf(1).toFloat(), Double.valueOf(0.0).toFloat(), Float.valueOf(3f).toFloat()) // Inlined: Calling the float overload
setToRotation(90.degress, 1f, 0f, 3f) // Optimized at compile-time, calling the float overload. 
setToRotation((90 * PI / 180.0), 1f, 0f, 3f) // After inline-class expansion
```

So in the end following that it ends using primitives that are optimized fine by all the targets.

Reply:
but this means that if I pass a vector, it gets unwrapped, then the x,y,z must be wrapped again as a vector in order to normalise, or we have to do the normalisation calculation in the rotation method.

which do you prefer

Reply:
ok, inlined the calculations